aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -653,7 +653,7 @@ Tensor& __irshift__(Tensor& self, Scalar other) {
 }
 
 template <typename Stub>
-Tensor& comparison_op_out(Tensor& result, const Tensor& self, const Tensor& other, Stub& stub) {
+Tensor& comparison_op_out(Tensor& result, const Tensor& self, const Tensor& other, Stub& stub, bool promote_output_types = true) {
```

Comment:
(still a WIP and subject to change- this is my attempt to special-case torch.ne)